src/test/java/core/basesyntax/strategy/OperationStrategyProviderTest.java

```diff
@@ -0,0 +1,55 @@
+package core.basesyntax.strategy;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.InventoryService;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class OperationStrategyProviderTest {
+    private InventoryService inventoryService;
+    private OperationStrategyProvider strategyProvider;
+
+    @BeforeEach
+    void setUp() {
+        inventoryService = new InventoryService();
+        strategyProvider = new OperationStrategyProvider(
+                Map.of(
+                        FruitTransaction.OperationType.SUPPLY, new SupplyOperationHandler(),
+                        FruitTransaction.OperationType.RETURN, new ReturnOperationHandler(),
+                        FruitTransaction.OperationType.ADD,
+                        new BalanceOperationHandler(inventoryService)
```

Comment:
```suggestion
                        new BalanceOperationHandler()
```
Can you avoid this dependency?